controller/aks-cluster-config-handler.go

```diff
@@ -184,7 +185,25 @@ func (h *Handler) recordError(onChange func(key string, config *aksv1.AKSCluster
 	return func(key string, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfig, error) {
 		var err error
 		var message string
-		config, err = onChange(key, config)
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// AKS config is being deleted
+			if config == nil {
+				return nil
+			}
+
+			conf, err := h.aksCC.Get(config.Namespace, config.Name, metav1.GetOptions{})
```

Comment:
if conf == nil {
    return fmt.Errorf("EKSClusterConfig %s/%s not found", config.Namespace, config.Name)
}